modules/imgcodecs/src/grfmt_webp.cpp

```diff
@@ -124,28 +81,51 @@ bool WebPDecoder::readHeader()
     }
 
     WebPBitstreamFeatures features;
-    if (VP8_STATUS_OK == WebPGetFeatures(header, sizeof(header), &features))
+    if (VP8_STATUS_OK < WebPGetFeatures(header, sizeof(header), &features)) return false;
+
+    m_has_animation = features.has_animation == 1;
+
+    if (m_has_animation)
     {
-        CV_CheckEQ(features.has_animation, 0, "WebP backend does not support animated webp images");
+        fs.seekg(0, std::ios::beg); CV_Assert(fs && "File stream error");
+        data.create(1, validateToInt(fs_size), CV_8UC1);
+        fs.read((char*)data.ptr(), fs_size);
+        CV_Assert(fs && "Can't read file data");
+        fs.close();
 
-        m_width  = features.width;
-        m_height = features.height;
+        CV_Assert(data.type() == CV_8UC1); CV_Assert(data.rows == 1);
 
-        if (features.has_alpha)
-        {
-            m_type = CV_8UC4;
-            channels = 4;
-        }
+        WebPData webp_data;
+        webp_data.bytes = (const uint8_t*)data.ptr();
+        webp_data.size = data.total();
+
+        WebPAnimDecoderOptions dec_options;
+        WebPAnimDecoderOptionsInit(&dec_options);
+
+        if(m_use_rgb)
```

Comment:
is the line below better for readability?
```
dec_options.color_mode = m_use_rgb
    ? (features.has_alpha ? MODE_RGBA : MODE_RGB)
    : (features.has_alpha ? MODE_BGRA : MODE_BGR);
```
@vrabaud , @asmorkalov

Reply:
I'd say the second one.